lib/Yield/Internal/Verify.cpp

```diff
@@ -0,0 +1,72 @@
+/// \file Verify.cpp
+/// \brief
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/Yield/Internal/BasicBlock.h"
+#include "revng/Yield/Internal/Function.h"
+#include "revng/Yield/Internal/Instruction.h"
+#include "revng/Yield/Internal/Tag.h"
+
+bool yield::Tag::verify(model::VerifyHelper &VH) const {
+  if (Type == TagType::Invalid)
+    return VH.fail("The type of this tag is not valid.");
+  if (FromPosition == std::string::npos)
+    return VH.fail("This tag doesn't have a starting point.");
+  if (ToPosition == std::string::npos)
+    return VH.fail("This tag doesn't have an ending point.");
+  if (FromPosition >= ToPosition)
+    return VH.fail("This tag doesn't have a positive length.");
+
+  return true;
+}
+
+bool yield::Instruction::verify(model::VerifyHelper &VH) const {
+  if (Address.isInvalid())
+    return VH.fail("An instruction has to have a valid address.");
+  if (Raw.empty())
+    return VH.fail("A raw view of an instruction cannot be empty.");
+  if (Bytes.empty())
+    return VH.fail("An instruction has to be at least one byte big.");
+
+  // TODO: Tags might need special verification as well.
+  // Their order, the fact that they don't overlap in undesirable ways, etc.
```

Comment:
If you switch to a `SortedVector` of elements sorted by `Start, End, Type` you have no problems with ordering by construction!

Reply:
I actually not sure whether we need that or not. I can't think of a case where there are two tags beginning and ending at the same points.

Reply:
ACK, the  consider implementing a custom `operator<`.

Reply:
What a weird way to write "the spaceship operator" /s

Reply:
What do you think?
https://github.com/revng/revng/pull/179/commits/a9a153ef846bdce867a44a0baa2ced49f31ce148 and https://github.com/revng/revng/pull/179/commits/d03f75b9360e0862c498787af8908a1d304c2de1